src/Machine.js

```diff
@@ -584,6 +614,192 @@ class Machine {
     }
   }
 
+  /**
+   * Create folder if path exists
+   * @param {string} dirPath path to directory
+   * @private
+   */
+  _mkdirSync(dirPath) {
+    try {
+      fs.mkdirSync(dirPath);
+    } catch (err) {
+      // if it is not "Exist error" then log it
+      if (err.code !== 'EEXIST') this.logger.error(err);
+    }
+  }
+
+  /**
+   * Create all missing folders in the current path
+   * @param {string} dirPath path to directory
+   * @private
+   */
+  _mkdirpSync(dirPath) {
+    const parts = dirPath.split(/[\\|\/]/);
+
+    // For every part of our path, call our wrapped _mkdirSync()
+    // on the full path until and including that part
+    for (let i = 1; i <= parts.length; i++) {
+      this._mkdirSync(path.join.apply(null, parts.slice(0, i)));
+    }
+  }
+
+  /**
+   * Create file with fileName path
+   * @param {string} fileName path to file
+   * @param {string} fileContent Content of the file
+   * @private
+   */
+  _createFile(fileName, fileContent) {
+    try {
+      fs.writeFileSync(fileName, fileContent);
+    } catch (err) {
+       this.logger.error(err);
+    }
+  }
+
+  /**
+   * @typedef {Object} NormalizedPath
+   * @property {string} fileName
+   * @property {string} dirPath
+   */
+
+  /**
+   * Transform url or github link to path and filename
+   * It is important, that path and filename are unique,
+   * because collision can break the build
+   * @param {string} link link to the file
+   * @return {NormalizedPath} folder and name, where cache file can be found
+   * @private
+   */
+ _normalizePath(link) {
+    const ghRes = GithubReader._parse(link);
+    if (ghRes !== false) {
+      return this._normalizeGithubPath(ghRes);
+    }
+    if (new HttpReader().supports(link)) { // CHANGE THIS
+      return this._normalizeHttpPath(link);
+    }
+  }
+
+  /**
+   * Transform github link to path and filename
+   * @param {string} link link to the file
+   * @return {NormalizedPath} folder and name, where cache file can be found
+   * @private
+   */
+ _normalizeHttpPath(httpLink) {
```

Comment:
1. Please don't invent a bicycle and use the standard way to parse URLs here.
2. URL normalization is not what you are doing here, https://en.wikipedia.org/wiki/URL_normalization